### ISSUE TITLE
Data Extraction fails when creating a Dev environment

### ISSUE THREAD

**Issue by MichaelNielsenDK** [vendrhub/vendr-deploy#11]
When creating a Dev environment, data extraction will fail with the error below.

`Object reference not set to an instance of an object.`
```
{
  "Date": "2021-12-20T08:40:30.5500524+00:00",
  "Message": "Object reference not set to an instance of an object.",
  "Ex": {
    "ClassName": "System.NullReferenceException",
    "Message": "Object reference not set to an instance of an object.",
    "Data": null,
    "InnerException": null,
    "HelpURL": null,
    "StackTraceString": "   at Vendr.Common.Events.EventBus.Dispatch[T](T evt)\r\n   at Vendr.Core.Models.AggregateBase`3.Validate(IValidationEvent validationEvent)\r\n   at Vendr.Core.Models.Store.SetName(String name, String alias)\r\n   at Vendr.Core.Models.Store.<>c__DisplayClass66_0.<Create>b__0(Store x)\r\n   at Vendr.Core.Models.AggregateBase`3.QuietSet(Action`1 actions)\r\n   at Vendr.Core.Models.Store.Create(IUnitOfWork uow, StoreState state, String alias, String name, Boolean autoPopulateStore)\r\n   at Vendr.Core.Models.Store.Create(IUnitOfWork uow, Guid id, String alias, String name, Boolean autoPopulateStore)\r\n   at Vendr.Deploy.Connectors.ServiceConnectors.VendrStoreServiceConnector.Pass1(ArtifactDeployState`2 state, IDeployContext context)\r\n   at Vendr.Deploy.Connectors.ServiceConnectors.VendrStoreServiceConnector.Process(ArtifactDeployState`2 state, IDeployContext context, Int32 pass)\r\n   at Umbraco.Deploy.Infrastructure.Environments.CurrentEnvironment.ProcessArtifacts(IEnumerable`1 arts, IDeployContext context, CancellationToken token)",
    "RemoteStackTraceString": null,
    "RemoteStackIndex": 0,
    "ExceptionMethod": null,
    "HResult": -2147467261,
    "Source": "Vendr.Common",
    "WatsonBuckets": null
  },
  "Sender": "Umbraco.Deploy.Infrastructure.Work.WorkItems.DiskReadWorkItem"
}
```

Umbraco 9.0.1
Vendr 2.0.3
Vendr.Deploy 2.0.1

Running a data extraction from the Umbraco backend will just result in the same error. 

But there is a work around to it:
Go into \site\wwwroot\umbraco\Deploy and manually run echo > deploy. You'll get an error saying "The environment is busy with another deployment", but this is an Umbraco Deploy bug, and if you just wait a little longer, you'll get a deploy-complete message.

**Comment by mattbrailsford** [comment]
Hey @MichaelNielsenDK I think this might be the same issue as here https://github.com/vendrhub/vendr-deploy/issues/10

Ultimately It's an Umbraco Deploy issue, magnified a little with some caching we do. Ultimately Deploy is running too soon before Vendr has been given chance to initialize. Because of this, it causes some things to be null which Vendr was then caching. We currently have a beta build on the unstable feed that removes this caching which means subsequent deploys via the back office will succeed, but the initial deploy will still fail.

I've reported this to HQ and I believe a PR has been merged into Deploy but I'm not sure what it's release cycle will be.
